targeted/WordPress User Enumeration.js

```diff
@@ -0,0 +1,204 @@
+
+// Print statements using script name
+function logger() {
+    print("[" + this["zap.script.name"] + "] " + arguments[0]);
+}
+
+/**
+ * Check for user enumeration on WordPress through author archives.
+ * 
+ * A function which will be invoked against a specific "targeted" message.
+ * 
+ * @param msg - the HTTP message being acted upon. This is an HttpMessage object.
+ */
+function invokeWith(msg) {
+
+    var url = msg.getRequestHeader().getURI().toString();
+    var alertName = "Wordpress Username Enumeration";
+    var alertDesc = "The username for user login has been exposed through author archives. This can allow for bruteforcing the password for the respective username and gain access to the admin dashboard.";
+    var alertSol = "Make sure that URLs with query parameter " + url + "/?author={integer} and " + url + "/wp-json/wp/v2/users are not accessible publicly.";
+    var alertReference = "https://www.getastra.com/blog/cms/wordpress-security/stop-user-enumeration/";
+    var cweId = 203; // Observable Discrepancy
+    var wascId = 13; // Information Leakage
+
+    // To check if script is running
+    logger("Testing the following URL -> " + url);
+    // Call function to check enumeration using /wp-json/wp/v2/users endpoint
+    var jsonAuthors = archiveJson(msg);
+
+    // Call function to check enumeration using /?author={i} endpoint
+    var archiveAuthors = userEnumerate(msg);
+
+    // If any username(s) found
+    if (jsonAuthors.length) {
+        logger("Vulnerable to user enumeration.");
+        var alertEvidence1 = "Found author(s) -> " + jsonAuthors.join(", ");
+        // Raise alert
+        raiseAlert(
+            pluginid,
+            3, // risk: 0: info, 1: low, 2: medium, 3: high
+            4, // confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+            alertName,
+            alertDesc,
+            alertEvidence1,
+            alertSol,
+            alertReference,
+            cweId,
+            wascId,
+            msg,
+            url + "/wp-json/wp/v2/users"
+        );
+    }
+
+    if (archiveAuthors.length) {
+        logger("Vulnerable to user enumeration.");
+        var alertEvidence = "Found author(s) -> ";
+        for (var i in archiveAuthors) {
+            alertEvidence += archiveAuthors[i].username + ", ";
+        }
+        url = url + "/?author=" + archiveAuthors[0].iterator;
+        // Raise alert
+        raiseAlert(
+            pluginid,
+            3, // risk: 0: info, 1: low, 2: medium, 3: high
+            3, // confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+            alertName,
+            alertDesc,
+            alertEvidence,
+            alertSol,
+            alertReference,
+            cweId,
+            wascId,
+            msg,
+            url
+        );
+    }
+    logger("Script run completed.");
+}
+
+/**
+ * 
+ * @param {HttpMessage} msg 
+ * @returns {Object.object}
+ */
```

Comment:
This jsdoc is kind of pointless if the param/returns aren't going to actually be further defined.